osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs

```diff
@@ -38,12 +45,14 @@ public void TestSongBar()
             AddStep("set beatmap", () =>
             {
                 var beatmap = CreateAPIBeatmap(Ruleset.Value);
+
                 beatmap.CircleSize = 3.4f;
                 beatmap.ApproachRate = 6.8f;
                 beatmap.OverallDifficulty = 5.5f;
                 beatmap.StarRating = 4.56f;
                 beatmap.Length = 123456;
                 beatmap.BPM = 133;
+                beatmap.OnlineID = CreateSampleBeatmap().OnlineID;
```

Comment:
The test doesn't cover the failure. When I check out https://github.com/ppy/osu/pull/24601/commits/5009fd379421f928bca35028b00e6b88374bdb5e, there's no crash.

There _is_ one if I do this:

```diff
diff --git a/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs b/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
index d52b453185..e0444b6126 100644
--- a/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
+++ b/osu.Game.Tournament.Tests/Components/TestSceneSongBar.cs
@@ -14,6 +14,7 @@ namespace osu.Game.Tournament.Tests.Components
     public partial class TestSceneSongBar : TournamentTestScene
     {
         private SongBar songBar = null!;
+        private TournamentBeatmap ladderBeatmap = null!;
 
         [SetUpSteps]
         public override void SetUpSteps()
@@ -22,9 +23,10 @@ public override void SetUpSteps()
 
             AddStep("setup picks bans", () =>
             {
+                ladderBeatmap = CreateSampleBeatmap();
                 Ladder.CurrentMatch.Value!.PicksBans.Add(new BeatmapChoice
                 {
-                    BeatmapID = CreateSampleBeatmap().OnlineID,
+                    BeatmapID = ladderBeatmap.OnlineID,
                     Team = TeamColour.Red,
                     Type = ChoiceType.Pick,
                 });
@@ -52,7 +54,7 @@ public void TestSongBar()
                 beatmap.StarRating = 4.56f;
                 beatmap.Length = 123456;
                 beatmap.BPM = 133;
-                beatmap.OnlineID = CreateSampleBeatmap().OnlineID;
+                beatmap.OnlineID = ladderBeatmap.OnlineID;
 
                 songBar.Beatmap = new TournamentBeatmap(beatmap);
             });

```

tl;dr: `CreateSampleBeatmap()` doesn't return the same `OnlineID` every time.

Reply:
Ugh, that's a final change I made to fix the online ID always being 1. Always gets me (refactoring after confirming coverage).

Your change looks more correct.